src/App.tsx

```diff
@@ -15,36 +16,110 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  name = 'name',
+  length = 'length',
+}
+
 export const App: React.FC = () => {
+  const [visibleGoods, setVisibleGoods] = useState<string[]>(goodsFromServer);
+  const [sortField, setSortField] = useState<SortType | ''>('');
+  const [isReversed, setIsReversed] = useState(false);
+
+  const sortAlphabetically = () => {
+    const sortedGoods = [...visibleGoods].sort((a, b) => a.localeCompare(b));
+
+    setVisibleGoods(sortedGoods);
+    setSortField(SortType.name);
+  };
+
+  const sortByLength = () => {
+    const sortedGoods = [...visibleGoods].sort((a, b) => a.length - b.length);
+
+    setVisibleGoods(sortedGoods);
+    setSortField(SortType.length);
+  };
+
+  const reset = () => {
+    setVisibleGoods(goodsFromServer);
+    setSortField('');
+    setIsReversed(false);
+  };
+
+  const reverseList = () => {
+    setVisibleGoods([...visibleGoods].reverse());
+    setIsReversed(!isReversed);
+  };
+
+  React.useEffect(() => {
+    if (sortField) {
+      const sortedGoods = [...visibleGoods].sort((good1, good2) => {
+        switch (sortField) {
+          case SortType.name:
+            return good1.localeCompare(good2);
+          case SortType.length:
+            return good1.length - good2.length;
+          default:
+            return 0;
+        }
+      });
+
+      if (isReversed) {
+        sortedGoods.reverse();
+      }
+
```

Comment:
move all this logic to some function (getVisibleGoods)

and then just call it
`const visibleGoods = getVisibleGoods(.......)`
and use this var instead of the  setVisibleGoods state